src/components/Image/BlankImage.tsx

```diff
@@ -15,7 +15,7 @@ type Props = {
 type ElementProps = Omit<HTMLAttributes<HTMLElement>, keyof Props>
 
 export const BlankImage: VFC<Props & ElementProps> = ({
-  alt = 'BlankImage',
+  alt = '',
```

Comment:
空文字だとそれはそれで意味があることになりそう。
undefinedだとダメそうですかね？

Reply:
プロダクト内で BlankImage が使われる場合「ファイルが選択されていません。」とか「{オブジェクト}がありません。」みたいな文章とともに表示されるはずなので、この画像は「装飾」扱いの空文字が適切かと思います!
`undefined` だと alt なしになりますよね?

Reply:
@uknmr 
> プロダクト内で BlankImage が使われる場合「ファイルが選択されていません。」とか「{オブジェクト}がありません。」みたいな文章とともに表示されるはず

実態としてそのように利用されることが多いとしても、あくまでこのコンポーネントは空で有ることを表す画像であるため、実際にどのように利用されるかはわからないかなと。
全員が仕様を把握して便利につかってくれるなら良いんですが `画像だけどデフォルトでaltが設定されている` ことがわからない気がします（例えばsmarthrロゴなら逆にalt入ってても違和感はないです）
また例で出されているパターンの場合でも画像自体を表すaltを入れるパターンも存在するかと　（例: 空のディレクトリを探す女性の画像 みたいな）

Reply:
> 実際にどのように利用されるかはわからない

わからないから、現実的に利用される状況に近い「空」が適切なデフォルトなのでは、と考えています。
Story で使われる場合の alt は「空のディレクトリを探す女性の画像」が適切だと思います。

Reply:
んーちょっと意図が違っていまして。

- 現実的に利用される状況が多く想定される空をデフォルトにしたい
  - わかる
- 実際に空をデフォルトにした場合
  - altを意識せずに利用することになる
  - ↑が問題だと思っています

実際空文字をデフォにしてしまって問題は起こらないだろうと思っていますが（上書きしたければできますし）、結果開発者がaltを意識しなくなるのはいややなぁくらいの印象です。
意思をもって空文字を設定してほしい、という意見です。（意図せず使えたほうが良いやんという意見ももちろんわかりますが...）

Reply:
「意識してほしい」も「意思をもって空文字を設定してほしい」もわかるんですが、それはコンポーネントの役割ではないと考えます。
すべての HTML や CSS を本当は意識して書いてほしいんだけど、限界があるため集約して提供している、というのはコンポーネントが持つ側面の一つだと思います。

初期値 `undefined` にした場合、基本的には意識せずに利用され alt なしの画像になるはず。linter やその他対策で防ぐのであればそれは smarthr-ui とセットで使うように明示なり強制する必要がでてくるのでは。

Reply:
> すべての HTML や CSS を本当は意識して書いてほしいんだけど、限界があるため集約して提供している、というのはコンポーネントが持つ側面の一つだと思います

これも基本同意なんですが、 BlankImage というコンポーネントに対して何を期待するか、というところで認識が違っていそうかな、という印象です。

- smarthr-ui が styleなどの見た目や動作的なものを担保してくれるのは 🙆‍♂️  ですが、html的な意味まで担保するのはちょっと違うかな？と思います
  - 例えば特定のコンポーネント内で画像が利用されており、それらに対してaltなどを固定で設定するのはコンポーネント内の設置箇所などによって意味が固定されるため問題ないと思います
  - 今回のBlankImageは基本的に利用される場合はある程度想像されるとはいえ、画像単体で公開されているものであるため、利用方法を固定できていないのが引っかかっています
  - uknmrさんの意見を取り入れるとしたら BlankImageと表示する文字を一セットにしたコンポーネントを提供するべきな気がします
    - ↑の場合ならそのコンポーネント内で `<BlankImage alt="" />` がされていてOKかと
- 特にBlankImageは画像、つまりimgタグかソレに類するものを期待する気がするのでaltが初期設定されてるとなるとそこからも逸脱しちゃう気が
  - これもsmarthr-uiは完璧に守っているわけではないですが...

Reply:
W3Cの情報を見ると、以下のような情報がありました。

> Decorative images do not add information to the content of the page. For example, the information provided by an image may already be given by the adjacent text, or the image may be included to make the website more attractive.
> In such cases, null (empty) alt text should be provided (alt="") so that it can be ignored by assistive technologies such as screen readers. Specifying a value for this type of image may cause audio distortions in the screen reader output or distract the user if it differs from the topic of the adjacent text. Omission of the alt attribute is not an option, as some screen readers will display the filename of the image if no alt attribute is specified.
> 
> 装飾的な画像は、ページのコンテンツに情報を追加するものではありません。例えば、画像によって提供される情報は、隣接するテキストによって既に与えられている場合があり、また、画像はウェブサイトをより魅力的にするために含まれている場合があります。
> このような場合、スクリーン・リーダーなどの支援技術によって無視されるように、ヌル（空）のaltテキストを提供する必要があります（alt=""）。このタイプの画像に値を指定すると、スクリーン・リーダーの出力に音声の歪みが生じたり、隣接するテキストの話題と異なる場合にユーザーの注意が散漫になったりすることがあります。alt属性が指定されていない場合、画像のファイル名が表示されるスクリーン・リーダーもあるため、alt属性の省略は選択肢にはなりません。
> https://www.w3.org/WAI/tutorials/images/decorative/


undefinedにする考え方を否定するわけではないのですが、アクセシビリティの観点からaltを外すという選択肢は取りづらいのかも？と思いました。

Reply:
> html的な意味まで担保するのはちょっと違う

HTML まで担保するので価値があるし、HTML と CSS は表裏一体だと思います。極端な例えをすると、すべて div や span になると開発はとても大変です。

> BlankImage というコンポーネントに対して何を期待するか

名前的にも、実態的にも装飾でしかないので `alt=""` は、`alt="BlankImage"` や `alt={undefined}` として上書き前提にするよりは妥当だと考えます。

> BlankImageと表示する文字を一セットにしたコンポーネントを提供

これは賛成です!

Reply:
> HTML まで担保するので価値があるし、HTML と CSS は表裏一体だと思います。極端な例えをすると、すべて div や span になると開発はとても大変です。

これはうまく伝わっていなかった案件だと思うのですがコンポーネントとして意味が固定出来る箇所は担保していいと考えています。
コレも極端な例になりますが、例えばリストとそのリストの名前を表示するコンポーネントがあるとしてそれらをul, liなどで実装するのはsmarthr-uiが担保するべき箇所だと思いますが、リストの名称のデフォルトを `従業員一覧` にしてしまうのは違うのでは？という感じです。
(この場合は利用者が上書きできるから良いやん？とはならない気がします)

今回の場合、`BlankImage` なのでそもそも `なにかが空の場合の概念を表す画像` として smarthr-uiは提供していますが、実際の利用方法自体を縛っているわけではないので `alt=""` と意味づけするならば実際の想定される利用方法に寄り添う形のコンポーネントを提供するのが筋なのかなぁと。(単純に何かを探しているアイコンとして使いたい場合とかもあるかもですし)

 話にも出ていたテキストと組み合わせる形のコンポーネントとして提供し、BlankImage単体では提供しないとかが良いのかなぁと思っています

Reply:
> alt指定を必須

利用状況が想像できるので、開発側に alt の質を委ねたくないです。
ほぼ必須と同等の意味で `alt=""` にしている気持ち。

Reply:
> 利用状況が想像できるので、開発側に alt の質を委ねたくないです。

altの質を担保したいならそれこそ極論ですがaltを指定不可能にして `alt=""` になることを強制するべきな気がします。
それか↓か...

> テキストと組み合わせる形のコンポーネントとして提供し、BlankImage単体では提供しないとかが良いのかなぁと思っています。

Reply:
> altを指定不可能にし

ありだと思います。

> BlankImage単体では提供しない

BlankImage をそもそも消す案もあると思うので、別 issue でやりましょう!

Reply:
📝 まとめ

- BlankImageのaltオプションは廃止し、alt=""で固定する (この画像の意味を外部から変更できなくする)
  - つまり他にテキストなどと同時に使わなければ意味のない画像となることを強制する
- 上記状況に合わせてテキスト＋BlankImageなコンポーネントの提供を検討する
- そもそもBlankImageが消される可能性があるためそちらを検討してから。